config/config.go

```diff
@@ -1015,6 +1047,70 @@ func configFileHasDependencyBlock(configPath string, terragruntOptions *options.
 	return false, nil
 }
 
+const (
+	MetadataTerraform                   = "terraform"
+	MetadataTerraformBinary             = "terraform_binary"
+	MetadataTerraformVersionConstraint  = "terraform_version_constraint"
+	MetadataTerragruntVersionConstraint = "terragrunt_version_constraint"
+	MetadataRemoteState                 = "remote_state"
+	MetadataDependencies                = "dependencies"
+	MetadataDependency                  = "dependency"
+	MetadataDownloadDir                 = "download_dir"
+	MetadataPreventDestroy              = "prevent_destroy"
+	MetadataSkip                        = "skip"
+	MetadataIamRole                     = "iam_role"
+	MetadataIamAssumeRoleDuration       = "iam_assume_role_duration"
+	MetadataIamAssumeRoleSessionName    = "iam_assume_role_session_name"
+	MetadataInputs                      = "inputs"
+	MetadataLocals                      = "locals"
+	MetadataGenerateConfigs             = "generate"
+	MetadataRetryableErrors             = "retryable_errors"
+	MetadataRetryMaxAttempts            = "retry_max_attempts"
+	MetadataRetrySleepIntervalSec       = "retry_sleep_interval_sec"
+)
```

Comment:
Is there any way to avoid having to maintain this list by hand? It feels very redundant with the config definition and parsing, which already has too much duplication in it, and I'm worried that when we go to update one, we'll forget to update the other.

Reply:
Out of curiosity, before this PR, where were the names of the fields used in the `render-json` output coming from?

Reply:
In the current implementation, it is hardcoded during generation: https://github.com/gruntwork-io/terragrunt/blob/master/config/config_as_cty.go#L23

I was thinking to change this part too to use the same constants